packages/core/src/external/aws/lambda-logic/modify-resources.ts

```diff
@@ -0,0 +1,599 @@
+  };
+}
+
+export type SlimPractitioner = Omit<Practitioner, "name" | "qualification" | "address"> & {
+  name?: string | undefined;
+  qualification?: string | undefined;
+  address?: string | undefined;
+  reference?: Record<string, string>;
+};
+
+function slimPractitioner(res: Practitioner): SlimPractitioner {
+  const updRes = cloneDeep(res);
+  return {
+    ...updRes,
+    name: getNameString(res.name),
+    qualification: getLongestDisplay(res.qualification?.[0]?.code),
+    address: getAddressString(res.address),
+  };
+}
+
+export type SlimProcedure = Omit<Procedure, "name" | "status" | "bodySite"> & {
+  name?: string | undefined;
+  status?: string | undefined;
+  reference?: Record<string, string>;
+  bodySite?: string | undefined;
+};
+
+function slimProcedure(res: Procedure): SlimProcedure | undefined {
+  const name = getUniqueDisplaysString(res.code);
+  if (name?.includes("no data")) return undefined;
+
+  const bodySite = getUniqueDisplaysString(res.bodySite);
+  const updRes = cloneDeep(res);
+  delete updRes.code;
+  delete updRes.reasonCode; // TODO: #2510 - Introduce term server lookup here
+  delete updRes.report;
+  delete updRes.note;
+  return {
+    ...updRes,
+    name,
+    status: isUselessStatus(res.status) ? undefined : res.status,
+    bodySite: bodySite ?? undefined,
+  };
+}
+
+export type SlimDiagnosticReport = Omit<
+  DiagnosticReport,
+  "type" | "category" | "presentedForm" | "status"
+> & {
+  type?: string | undefined;
+  category?: string | undefined;
+  presentedForm?: string[] | undefined;
+  status?: string | undefined;
+  // reference?: Record<string, string>;
+  reference?: Record<string, string | object>;
+};
+
+function slimDiagnosticReport(res: DiagnosticReport): SlimDiagnosticReport | undefined {
+  const mostDescriptiveType = getLongestDisplay(res.code);
+  const allTypes = getUniqueDisplays(res.code);
+
+  if (allTypes) {
+    for (const type of allTypes) {
+      if (
+        REPORT_TYPES_BLACKLIST.includes(type) ||
+        REPORT_TYPES_BLACKLIST.some(blacklistedType => type.includes(blacklistedType))
+      ) {
+        return undefined;
+      }
+    }
+  }
+
+  const uniqueData = new Set<string>();
+  let reportType = mostDescriptiveType;
+
+  res.presentedForm?.forEach(form => {
+    if (form.data) {
+      const rawData = Buffer.from(form.data, "base64").toString("utf-8");
+      const cleanedData = cleanUpNote(rawData);
+      uniqueData.add(cleanedData);
+
+      if (cleanedData.toLowerCase().includes("appointment scheduling letter")) {
+        reportType = SCHEDULING_CALL;
+      } else if (
+        cleanedData.toLowerCase().includes("unable to reach") ||
+        cleanedData.toLowerCase().includes("left a message")
+      ) {
+        reportType = UNANSWERED_CALL;
+      } else if (cleanedData.toLowerCase().includes("administrative note")) {
+        reportType = ADMIN_NOTE;
+      } else if (
+        cleanedData.toLowerCase().includes("nonva note") &&
```

Comment:
sanity check: "nonva"?

Reply:
ya, i think it's just some facility that adds those useless notes..